axiom/nr_segment.h

```diff
@@ -647,4 +647,22 @@ extern void nr_segment_record_exception(nr_segment_t* segment,
                                         const char* error_message,
                                         const char* error_class);
 
+/*
+ * Purpose : Gets the child_ix of a segment.
+ *
+ * Paramas : 1. The pointer to the segment
+ *
+ * Returns : the child_ix, or 0 if passed NULL
+ *           (Note: 0 is a valid ix)
```

Comment:
I suggest changing the return type to `ssize_t` and return `-1` if passed `NULL`.

Reply:
I think I would rather the primary onux be on the caller to not pass NULL here (or know if they pass NULL). Otherwise, every time this function is called there will need to be a cast from `ssize_t` to `size_t`, after a `-1` check of course. So ultimately, it is extra code and in my opinion less readable.

Reply:
Returning a valid value in error condition may result in an unexpected behavior and this does not seem right. With `ssize_t` as return type the compiler will enforce the correct use of the returned value. I prefer trading off the need to consciously end explicitly casting over unexpected behavior.